### DISCUSSION TITLE
Emacs keybinding do not work properly at start and end of a file

### DISCUSSION THREAD

**Discussion by mikysett** [zed-industries/zed#17201] [Feature Requests]
I remapped the keybinding to get a more Emacs touch:
```json
      "ctrl-p": "editor::MoveUp",
      "ctrl-n": "editor::MoveDown",
```

Those works properly when I'm in a middle line of a file, but when I'm at the top `ctrl-p` opens search project files and (very annoying) `ctrl-n` will start to open new files at a very high speed.

I guess it's only about remapping those two also in some other contexts but all my tries miserably failed.
Also I wouldn't expect the binding context to change when reaching start/end of file.
Would be great to get unstuck on that.

PS: This issue is only relevant on Linux/Windows as Mac has `cmd-n` for new files.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
